File: producer-api-go-rest/internal/handlers/health.go
package handlers

import (
	"github.com/gin-gonic/gin"
)

type HealthHandler struct{}

func NewHealthHandler() *HealthHandler {
	return &HealthHandler{}
}

func (h *HealthHandler) HealthCheck(c *gin.Context) {
	c.JSON(200, gin.H{
		"status":  "healthy",
		"message": "Producer API is healthy",
	})
}

File: producer-api-rust/src/handlers/health.rs
use axum::{response::Json, routing::get, Router};
use serde_json::json;
use crate::service::EventProcessingService;

pub fn router() -> Router<EventProcessingService> {
    Router::new().route("/health", get(health_check))
}

async fn health_check() -> Json<serde_json::Value> {
    Json(json!({
        "status": "healthy",
        "message": "Producer API is healthy"
    }))
}

